tuf/client_rework/updater_rework.py

```diff
@@ -142,44 +142,38 @@ def updated_targets(targets: Dict, destination_directory: str) -> Dict:
 
         return updated_targets
 
-    def download_target(self, target: Dict, destination_directory: str):
+    def download_target(
+        self,
+        targetinfo: Dict,
+        destination_directory: str,
+        target_url: Optional[str] = None,
+    ):
         """
         This method performs the actual download of the specified target.
         The file is saved to the 'destination_directory' argument.
         """
+        if target_url is None:
+            target_url = self._default_target_url
 
-        temp_obj = None
-        file_mirror_errors = {}
-        file_mirrors = mirrors.get_list_of_mirrors(
-            "target", target["filepath"], self._mirrors
-        )
-
-        for file_mirror in file_mirrors:
-            try:
-                temp_obj = download.download_file(
-                    file_mirror, target["fileinfo"]["length"], self._fetcher
-                )
-                _check_file_length(temp_obj, target["fileinfo"]["length"])
-                temp_obj.seek(0)
-                _check_hashes_obj(temp_obj, target["fileinfo"]["hashes"])
-                break
-
-            except Exception as exception:  # pylint:  disable=broad-except
-                # Store the exceptions until all mirrors are iterated.
-                # If an exception is raised from one mirror but a valid
-                # file is found in the next one, the first exception is ignored.
-                file_mirror_errors[file_mirror] = exception
-
-                if temp_obj:
-                    temp_obj.close()
-                    temp_obj = None
-
-        # If all mirrors are iterated but a file object is not successfully
-        # downloaded and verifies, raise the collected errors
-        if not temp_obj:
-            raise exceptions.NoWorkingMirrorError(file_mirror_errors)
+        full_url = _build_full_url(target_url, targetinfo["filepath"])
 
-        filepath = os.path.join(destination_directory, target["filepath"])
+        temp_obj = None
+        try:
+            temp_obj = download.download_file(
+                full_url, targetinfo["fileinfo"]["length"], self._fetcher
+            )
+            _check_file_length(temp_obj, targetinfo["fileinfo"]["length"])
+            temp_obj.seek(0)
+            _check_hashes_obj(temp_obj, targetinfo["fileinfo"]["hashes"])
+
+        except Exception as e:
+            if temp_obj:
+                temp_obj.close()
+            # TODO: do we reraise a NoWorkingMirrorError or just
+            # let exceptions propagate?
+            raise exceptions.NoWorkingMirrorError({full_url: e}) from e
```

Comment:
(here and below) does `NoWorkingMirrorError` still make sense, given we're removing mirrors?